source/stdcpp/vector.d

```diff
@@ -750,10 +768,84 @@ extern(D):
         inout(T)[] as_array() inout pure nothrow @trusted @nogc             { return null; }
         ref inout(T) at(size_type i) inout pure nothrow @trusted @nogc      { data()[0]; }
     }
-    else
+    else version(CppRuntime_Gcc)
     {
-        static assert(false, "C++ runtime not supported");
+
+
+	//commented out for now,will be cleared later
+       // static assert(false, "C++ runtime not supported");
+	
+
+
+	//instantiating allocator to rebind and use allocate function is current problem
+	//constructor has a form 'this(U)(ref allocator!U) {}' which the only option in mind is to circularly reference
+	// which causes an error
+	allocator!value_type p;
+	
+	p.rebind!pointer _M_start;
+	p.rebind!pointer _M_finish;
+	p.rebind!pointer _M_end_of_storage;
+
+
+	// vector(n) constructor
+
+	this()(size_t __n)
```

Comment:
Note that this constructor is templated, something that is not possible in C++. So this does not mangle to the constructor as it should.
Try compiling the following:
```C++
#include <vector>

int main ()
{
    std::vector<int> vec;
    return 0;
}
```

And then `nm a.out`, you'll get the following (removed irrelevant output):
```
0000000000001481 W _ZNSt12_Destroy_auxILb1EE9__destroyIPiEEvT_S3_
000000000000121e W _ZNSt12_Vector_baseIiSaIiEE12_Vector_implC1Ev
000000000000121e W _ZNSt12_Vector_baseIiSaIiEE12_Vector_implC2Ev
00000000000011be W _ZNSt12_Vector_baseIiSaIiEE12_Vector_implD1Ev
00000000000011be W _ZNSt12_Vector_baseIiSaIiEE12_Vector_implD2Ev
0000000000001366 W _ZNSt12_Vector_baseIiSaIiEE13_M_deallocateEPim
0000000000001324 W _ZNSt12_Vector_baseIiSaIiEE17_Vector_impl_dataC1Ev
0000000000001324 W _ZNSt12_Vector_baseIiSaIiEE17_Vector_impl_dataC2Ev
00000000000013a0 W _ZNSt12_Vector_baseIiSaIiEE19_M_get_Tp_allocatorEv
00000000000011de W _ZNSt12_Vector_baseIiSaIiEEC1Ev
00000000000011de W _ZNSt12_Vector_baseIiSaIiEEC2Ev
000000000000126a W _ZNSt12_Vector_baseIiSaIiEED1Ev
000000000000126a W _ZNSt12_Vector_baseIiSaIiEED2Ev
00000000000013ef W _ZNSt16allocator_traitsISaIiEE10deallocateERS0_Pim
00000000000011fe W _ZNSt6vectorIiSaIiEEC1Ev
00000000000011fe W _ZNSt6vectorIiSaIiEEC2Ev
00000000000012bc W _ZNSt6vectorIiSaIiEED1Ev
00000000000012bc W _ZNSt6vectorIiSaIiEED2Ev
0000000000001421 W _ZSt8_DestroyIPiEvT_S1_
00000000000013b2 W _ZSt8_DestroyIPiiEvT_S1_RSaIT0_E
```

If you run c++filt on this:
```
0000000000001481 W void std::_Destroy_aux<true>::__destroy<int*>(int*, int*)
000000000000121e W std::_Vector_base<int, std::allocator<int> >::_Vector_impl::_Vector_impl()
000000000000121e W std::_Vector_base<int, std::allocator<int> >::_Vector_impl::_Vector_impl()
00000000000011be W std::_Vector_base<int, std::allocator<int> >::_Vector_impl::~_Vector_impl()
00000000000011be W std::_Vector_base<int, std::allocator<int> >::_Vector_impl::~_Vector_impl()
0000000000001366 W std::_Vector_base<int, std::allocator<int> >::_M_deallocate(int*, unsigned long)
0000000000001324 W std::_Vector_base<int, std::allocator<int> >::_Vector_impl_data::_Vector_impl_data()
0000000000001324 W std::_Vector_base<int, std::allocator<int> >::_Vector_impl_data::_Vector_impl_data()
00000000000013a0 W std::_Vector_base<int, std::allocator<int> >::_M_get_Tp_allocator()
00000000000011de W std::_Vector_base<int, std::allocator<int> >::_Vector_base()
00000000000011de W std::_Vector_base<int, std::allocator<int> >::_Vector_base()
000000000000126a W std::_Vector_base<int, std::allocator<int> >::~_Vector_base()
000000000000126a W std::_Vector_base<int, std::allocator<int> >::~_Vector_base()
00000000000013ef W std::allocator_traits<std::allocator<int> >::deallocate(std::allocator<int>&, int*, unsigned long)
00000000000011fe W std::vector<int, std::allocator<int> >::vector()
00000000000011fe W std::vector<int, std::allocator<int> >::vector()
00000000000012bc W std::vector<int, std::allocator<int> >::~vector()
00000000000012bc W std::vector<int, std::allocator<int> >::~vector()
0000000000001421 W void std::_Destroy<int*>(int*, int*)
00000000000013b2 W void std::_Destroy<int*, int>(int*, int*, std::allocator<int>&)
```

So as you can see , the C++ template emits the ctor, you just have to bind it.